new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java

```diff
@@ -628,29 +631,41 @@ public static String createMiiImageAndVerify(String miiImageNameBase,
 
       if (archiveAppsList.size() != 0 && archiveAppsList.get(0) != null) {
         assertTrue(archiveApp(defaultAppParams()
-                .srcDirList(archiveAppsList)));
+            .srcDirList(archiveAppsList)));
         //archive provided ear or war file
         String appName = archiveAppsList.get(0).substring(archiveAppsList.get(0).lastIndexOf("/") + 1,
-                appSrcDirList.get(0).lastIndexOf("."));
+            appSrcDirList.get(0).lastIndexOf("."));
 
         // build the archive list
         String zipAppFile = String.format("%s/%s.zip", ARCHIVE_DIR, appName);
         archiveList.add(zipAppFile);
 
       }
+
       if (buildAppDirList.size() != 0 && buildAppDirList.get(0) != null) {
         // build an application archive using what is in resources/apps/APP_NAME
-        assertTrue(buildAppArchive(defaultAppParams()
-                        .srcDirList(buildAppDirList)),
-                String.format("Failed to create app archive for %s", buildAppDirList.get(0)));
-
-        // build the archive list
-        String zipFile = String.format("%s/%s.zip", ARCHIVE_DIR, buildAppDirList.get(0));
-        archiveList.add(zipFile);
+        String zipFile = "";
+        if (oneArchiveContainsMultiApps) {
+          assertTrue(buildAppArchive(defaultAppParams()
+                  .srcDirList(buildAppDirList)),
+              String.format("Failed to create app archive for %s", buildAppDirList.get(0)));
+          zipFile = String.format("%s/%s.zip", ARCHIVE_DIR, buildAppDirList.get(0));
+          // build the archive list
+          archiveList.add(zipFile);
+        } else {
+          for (String appName : buildAppDirList) {
+            assertTrue(buildAppArchive(defaultAppParams()
+                    .srcDirList(Collections.singletonList(appName))
+                    .appName(appName)),
+                String.format("Failed to create app archive for %s", appName));
+            zipFile = String.format("%s/%s.zip", ARCHIVE_DIR, appName);
+            // build the archive list
+            archiveList.add(zipFile);
+          }
+        }
       }
```

Comment:
why do we need to generate zip for each app if multiple app names are provided? it will zip all of them into one zip file with name of first provided app name dir.

Reply:
I need each app has one ear file. In the current logic in develop branch, it will build the multiple apps in one ear file. It will not work for me.